typescript/cli/src/registry/MergedRegistry.ts

```diff
@@ -0,0 +1,154 @@
+  public readonly isDryRun: boolean;
+
+  constructor({ registryUris, logger, isDryRun }: MergedRegistryOptions) {
+    logger ||= rootLogger.child({ module: 'MergedRegistry' });
+    super({ uri: '__merged_registry__', logger });
+
+    if (!registryUris.length)
+      throw new Error('At least one registry URI is required');
+
+    this.registries = registryUris.map((uri, index) => {
+      // If not provided, allow the GithubRegistry to use its default
+      if (isHttpsUrl(uri)) {
+        return new GithubRegistry({ uri, logger: logger!.child({ index }) });
+      } else {
+        return new LocalRegistry({ uri, logger: logger!.child({ index }) });
+      }
+    });
+
+    this.isDryRun = !!isDryRun;
+  }
+
+  async listRegistryContent(): Promise<RegistryContent> {
+    const results = await Promise.all(
+      this.registries.map((registry) => registry.listRegistryContent()),
+    );
+    return results.reduce((acc, content) => objMerge(acc, content), {
+      chains: {},
+      deployments: {},
+    });
+  }
+
+  async getChains(): Promise<Array<ChainName>> {
+    return objKeys(await this.getMetadata);
+  }
+
+  async getMetadata(): Promise<ChainMap<ChainMetadata>> {
+    const results = await Promise.all(
+      this.registries.map((registry) => registry.getMetadata()),
+    );
+    return results.reduce((acc, content) => objMerge(acc, content), {});
+  }
+
+  async getChainMetadata(chainName: ChainName): Promise<ChainMetadata | null> {
+    return (await this.getMetadata())[chainName] || null;
+  }
+
+  async getAddresses(): Promise<ChainMap<ChainAddresses>> {
+    const results = await Promise.all(
+      this.registries.map((registry) => registry.getAddresses()),
+    );
+    return results.reduce((acc, content) => objMerge(acc, content), {});
+  }
+
+  async getChainAddresses(
+    chainName: ChainName,
+  ): Promise<ChainAddresses | null> {
+    return (await this.getAddresses())[chainName] || null;
+  }
+
+  async addChain(chain: {
+    chainName: ChainName;
+    metadata?: ChainMetadata;
+    addresses?: ChainAddresses;
+  }): Promise<void> {
+    return this.multiRegistryWrite(
+      async (registry) => await registry.addChain(chain),
+      'adding chain',
+    );
+  }
+
+  async updateChain(chain: {
+    chainName: ChainName;
+    metadata?: ChainMetadata;
+    addresses?: ChainAddresses;
+  }): Promise<void> {
+    return this.multiRegistryWrite(
+      async (registry) => await registry.updateChain(chain),
+      'updating chain',
+    );
+  }
+
+  async removeChain(chain: ChainName): Promise<void> {
+    return this.multiRegistryWrite(
+      async (registry) => await registry.removeChain(chain),
+      'removing chain',
+    );
+  }
+
+  async addWarpRoute(config: WarpCoreConfig): Promise<void> {
+    return this.multiRegistryWrite(
+      async (registry) => await registry.addWarpRoute(config),
+      'adding warp route',
+    );
+  }
+
+  protected async multiRegistryWrite(
```

Comment:
I would expect in practice users only want to write to 1 registry at a time (and probably local registry)

Reply:
Not sure I agree but anyway we don't have to decide that detail right now. It's skipping the GithubRegistry writes atm